Jenkinsfile

```diff
@@ -19,11 +19,20 @@ def mavenEnv(Map params = [:], Closure body) {
       timeout(120) {
         withChecks(name: 'Tests', includeStage: true) {
           infra.withArtifactCachingProxy {
-            withEnv([
-              'JAVA_HOME=/opt/jdk-' + params['jdk'],
-              "MAVEN_ARGS=${env.MAVEN_ARGS != null ? MAVEN_ARGS : ''} -B -ntp -Dmaven.repo.local=${WORKSPACE_TMP}/m2repo"
+            cache(
```

Comment:
I don't know how this will behave with all the parallel branches and whether e.g. setting `skipSave` will be needed on the plugin branches (would mean test only deps would be missed)

Reply:
I wonder if `skipSave` shouldn't be `true` on PRs for safety reasons?

Reply:
cache is per branch IIUC and master is used as a seed for PRs so they aren't slow but need to double check, 
tbh probably unneeded to save PRs